core/web-assets/src/main/assets/js/apps/add-to-ksc/index.js

```diff
@@ -157,9 +198,39 @@ angular.module('onms-ksc', [
                      $scope.hasFlows = true;
             } else {
               $scope.flowGraphUrl = null;
-            }
+            } 
         }
-
       });
   };
+  // When graph search doesn't yield any results, set/reset noMatchingGraphs
+  $scope.$on('handleNoMatchingGraphsFound', function () {
+      $scope.nomatchingGraphs = graphSearchFactory.noMatchingGraphs;
+  });
+}])
+.controller('graphSearchCtrl', ['$scope', '$timeout', '$filter', '$attrs', '$element', 'graphSearchFactory', function($scope, $timeout, $filter, $attrs, $element, graphSearchFactory) {
+
+  let graphName = $attrs.graphname;
+  let graphTitle = $attrs.graphtitle;
+  // Update service with graphname and graphtitle.
+  graphSearchFactory.initialize(graphName, graphTitle);
+  $scope.enableGraph = true;
+  // Handle search query update and enable graphs with matching search query.
+  $scope.$on('handleSearchQuery', function () {
+    let searchQuery = graphSearchFactory.searchQuery;
+    // Filter on graphName or graphTitle.
+    let matchingElements = $filter('filter')([graphName, graphTitle], searchQuery);
+    if (matchingElements && matchingElements.length) {
+      $scope.enableGraph = true;
+      if(searchQuery) {
+        // Send event for the specific div to check if they are in viewport.
```

Comment:
Note that below code is small hack to get divs to render when they are in viewport.  I tried using  angular `$viewContentLoaded` but that doesn't work.  So I didn't find more deterministic way to handle this.

Reply:
Looks like `$timeout` is the default option here for what I want to achieve ( run after the DOM has been manipulated by Angular)  as I m triggering this event from controller.

https://stackoverflow.com/a/17303759/5216060